### ISSUE TITLE
Questions regarding Hyperopt

### ISSUE THREAD

**Issue by Vildnex** [freqtrade/freqtrade#3500]
## Describe your environment

  * Operating system: Manjaro 20.0.3
  * Python Version: 3.8 (`python -V`)
  * CCXT version: 1.30.2 (`pip freeze | grep ccxt`)
  * Freqtrade Version: develop-79933060 (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question

I do have 2 questions if it's possible:

1. In terms of running Hyperopt what going to be a better choice? SharpeHyperOptLossDaily or SharpeHyperOptLoss? Because so far as I read in the documentation is just saying that the first one is going to optimize for daily returns and the other one for the whole period. Which one should be better, or what should I take into consideration when I will choose one over the other?

2. What should be a normal time range for this optimization? 
In my strategy, I'm winging a 5min timeframe, so base on that I'm not sure how long should be the optimizing period to be.
For example, I've tried a period of 61 days (which I think it's quite a small number and takes something like 2-3 days to be finished with 10k epochs) but it's eating something like 16-20gb of ram (maybe a bit more). I've also tried a 665 days period (which in terms of the time range for me sounds more reasonable because I have more data) but this would take around 150-180gb of RAM and around of few weeks' time in order to be done).
So my question would be, what would be a reasonable time range in days for optimizing a 5min strategy? Also, I'm not sure if it's important but I'm using around of 200 pairs X/USDT for this training.

**Comment by xmatthias** [comment]
1) it strongly depends (mainly on the timerange / timeframe) ...
the 2 don't give each other too much - one uses the same returns but per day, the other uses them raw. results will differ - but there is no "better" or "worse" - they're just slightly different.

2)
the main problem i think is using 200 pairs ... if you drill it down ti some 30-40 pairs (a realistic number...) - the ram requirements will not be that high.
The ram can easily be explained as numpairs * days * cores.
Also - if you're not filtering out certain pairs, you'll end up with a strategy that has a few strong pairs with unrealistic gains - and the rest of the pairs without trades, or with losses - but all made up by the few pairs which got lucky trades of a few 100%.

Ideally, you'd also not want to run it for 10k epochs - but instead run it multiple times for smaller epoch sizes. Keep in mind that it's Bayesian search - so it may descent on a local minimum so it won't find another minimum anymore.

**Comment by Vildnex** [comment]
Ok... But regarding the answer to the second question...

1. How can I filter the pairs? I mean how can I know if thows pairs are ok with no unrealistic gain or lose?

2. If 10k is too much and I should run it multiple times, then what kind of number would be more realistic? 1k? 5k?

**Comment by xmatthias** [comment]
we usually reccomend 1k or so ...
everyhing beyond that will not significantly improve the results.

you can use test-pairlist to with one of the filters we have available.
Also, once you transfer your parameters from hyperopt to strategy - you'll want to backtest it again to confirm you did everything correctly - if one pair has 90% of the profits, look at that pair VERY carefully - as it's most likely a problematic pair.

**Comment by hroff-1902** [comment]
citing today's posts in Slack:

"It's usually useless to run hyperopt for more than 500-700-1000 epochs. Hyperopt uses bayesian search, which quickly descends to a local minimum of the objective function, it's not grid search in the hyperspace, which requires full coverage of the complete hyperspace with many epochs.

So instead of running hyperopt for 10000 epochs at once, you can better run it 20 times for 500 epochs or 10 times for 1000 epochs -- it will find you 20 or 10 different local minimums of the loss function, which you can then select manually from based on some external criteria (or try all of them with dry-run)."

**Comment by qkum** [comment]
> we usually reccomend 1k or so ...
> everyhing beyond that will not significantly improve the results.
> 
> you can use test-pairlist to with one of the filters we have available.
> Also, once you transfer your parameters from hyperopt to strategy - you'll want to backtest it again to confirm you did everything correctly - if one pair has 90% of the profits, look at that pair VERY carefully - as it's most likely a problematic pair.

If you trade BTC pairs you pretty much need to remove any pair that cost less than 100 sats. #1 the order books are huge and if you use market buy the spread is huge and you lose money/profit. # if a coin cost like below 50 sats the bot will like mathias say, the bot will do unending trades on that coin and make like 4-8% every trade even when it is impossible and then spit out a bad strategy to you in the end that wont work so well on most coins.

**Comment by hroff-1902** [comment]
> If you trade BTC pairs you pretty much need to remove any pair that cost less than 100 sats.

Right, can only agree with it. And that's what VolumePairList is intended for. The problem with it is that it's based on tocker's volume and historical tickers data is not available from exchanges. So VolumePairList cannot be used on historical data. The only way to simulate it is either by means of test-pairlist, on current ticker volumes or by filtering pairs manually.

**Comment by qkum** [comment]
> > If you trade BTC pairs you pretty much need to remove any pair that cost less than 100 sats.
> 
> Right, can only agree with it. And that's what VolumePairList is intended for. The problem with it is that it's based on tocker's volume and historical tickers data is not available from exchanges. So VolumePairList cannot be used on historical data. The only way to simulate it is either by means of test-pairlist, on current ticker volumes or by filtering pairs manually.

Aha, yeah, I did it manually after discovering where the crazy good hyperopt results came from.

**Comment by Vildnex** [comment]
Actually I'm using USDT as a pair, not BTC because I found it to be more stable in terms of it's value.

**Comment by qkum** [comment]
> Actually I'm using USDT as a pair, not BTC because I found it to be more stable in terms of it's value.

Good, 

Then you should not have the problems I talk about :)

**Comment by xmatthias** [comment]
I'll assume this as answered, as i don't think there are any unanswered questions from the original query.
 
Feel free to open a new issue (for new questions) or comment below if you still have questions regarding this question.
